### ISSUE TITLE
[Scoped] Symfony Polyfills are incorrectly prefixed

### ISSUE THREAD

**Issue by mkrauser** [rectorphp/rector#5617]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | e.g. v0.9.29              |
| Installed as   | prefixed Rector                           |

I got an exception, that the function get_debug_type is not defined. I tracked it down to the symfony polyfills incorrectly being prefixed.

The only solution I found to fix this, is to whitelist all bootstrap.php files and bootstrap80.php files within the polyfills. whitelist-global-functions does not work. This might also be an issue within php-scoper itself.

Here's the polyfill for get_debug_type defined: https://github.com/rectorphp/rector-prefixed/blob/f907c6d2101d9e34d027f59a43e5efa1db76790e/vendor/symfony/polyfill-php80/bootstrap.php#L50

If rector is executed on PHP < 8.0, and get_debug_type is called (either by rector- or by vendor-code), the function is undefined due to the namespace added [here](https://github.com/rectorphp/rector-prefixed/blob/f907c6d2101d9e34d027f59a43e5efa1db76790e/vendor/symfony/polyfill-php80/bootstrap.php#L3).

**Comment by TomasVotruba** [comment]
Hi, 

could you link to the code in https://github.com/rectorphp/rector-prefixed ?

**Comment by mkrauser** [comment]
Of course. I added it to the report

**Comment by TomasVotruba** [comment]
There is no `get_debug_type` in the linked file.

**Comment by TomasVotruba** [comment]
We'll need a reproducible repository on Github to solve this one

**Comment by mkrauser** [comment]
> We'll need a reproducible repository on Github to solve this one

I've updated the description and I'll try to implement a failing demo

**Comment by TomasVotruba** [comment]
Thanks!
Now it's hard to follow where the reply is. Could you use comments for next replies to keep the context from up to bottom?

**Comment by mkrauser** [comment]
Of course, I just wanted to keep everything in one place. 
It might take a few days before I find time to do that.

**Comment by TomasVotruba** [comment]
This looks like bug in php-scoper, as the function itself is prefixed correctly:

```php
// bug
if (!\function_exists('get_debug_type')) {
    function get_debug_type($value) : string
    {
        // correct
        return \RectorPrefix20210218\Symfony\Polyfill\Php80\Php80::get_debug_type($value);
    }
}
```

There are 2 ways to fix it:

- Right in php-scoper (better, but could take weeks to merge)
- Add a prefixer in our `scoper.php`: https://github.com/rectorphp/rector/blob/652459ad3be1004d971aac86cff36a002ad7d11a/scoper.php#L29

**Comment by mkrauser** [comment]
There is already an issue in php-scoper for this: https://github.com/humbug/php-scoper/issues/440

**Comment by TomasVotruba** [comment]
Closing here then, thanks for reference
